src/Core/Field.h

```diff
@@ -767,9 +767,10 @@ T & Field::get()
 #ifndef NDEBUG
     // Disregard signedness when converting between int64 types.
     constexpr Field::Types::Which target = TypeToEnum<NearestFieldType<ValueType>>::value;
-    assert(target == which
-           || (isInt64FieldType(target) && isInt64FieldType(which))
-           || target == Field::Types::Decimal64 /* DateTime64 fields */);
+    if (target != which
```

Comment:
This is probably going to slow down some things like `sumMap`.

Btw, just noticed that we now have Decimal64 here, but the author haven't added any justification as to why this check somehow doesn't apply to them :sigh:

Reply:
I think we should care about the clarity of errors in debug mode, rather than performance?)

Reply:
Somehow I didn't notice the NDEBUG around it, in debug mode it's OK.